src/test/java/com/google/javascript/gents/singleTests/inline_types.js

```diff
@@ -5,6 +5,6 @@ var /** !Foo */ foo = {};
 
 const /** string */ x = 'fruit';
 
-// TODO(#352): Support inline typing for functions.
-// input: function /** string */ f(/** number */ x) {return x + ' apples'}
-// output: function f(x: number): string {return x + ' apples'}
+function /** string */ f(/** number */ x, z, /** string */ y) {
```

Comment:
What'd happen if we mix function JSDoc (with or without `@param`s) with inline comments? I think your code handles that nicely, but could you add a test for the situation?